ts/state/selectors/conversations.ts

```diff
@@ -265,29 +248,64 @@ export const getConversationComparator = createSelector(getIntl, _getConversatio
 // tslint:disable-next-line: cyclomatic-complexity
 const _getLeftPaneLists = (
   sortedConversations: Array<ReduxConversationType>
-): {
-  conversations: Array<ReduxConversationType>;
-  contacts: Array<ReduxConversationType>;
-  globalUnreadCount: number;
-} => {
-  const conversations: Array<ReduxConversationType> = [];
-  const directConversations: Array<ReduxConversationType> = [];
+): Array<ReduxConversationType> => {
+  return sortedConversations.filter(conversation => {
+    if (conversation.isBlocked) {
+      return false;
+    }
+
```

Comment:
This could be cleaned up a bit 

```typescript
// ... code
// a private conversation not approved is a message request. Exclude them from the left pane lists
const isPrivateButUnapproved = conversation.isPrivate && !conversation.isApproved;
// a hidden contact conversation is only visible from the contact list, not from the global conversation list
const isPrivateButHidden =
      conversation.isPrivate &&
      conversation.priority &&
      conversation.priority <= CONVERSATION_PRIORITIES.default;
// When getting a contact from a linked device, before he sent a message, the approved field is false, but a createdAt is used as activeAt
const isPrivateUnapprovedButActive = isPrivateButUnapproved && !conversation.activeAt;

if (
  isPrivateButUnapproved ||
  isPrivateUnapprovedButActive ||
  isPrivateButHidden
) {
  // dont increase unread counter, don't push to convo list.
  return false;
}
// ... code
```

Reply:
I did a bunch of changes relative to this. Cleaned up the selector as we only need the ids and the count of them in the end.
Did the same thing for the search logic.